# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/vm/Interpreter.h
# Commit: 6ccaefced3df
# Full Hash: 6ccaefced3df63b297057b31a856527153801a71
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2020-01-13 15:45:24
# Description:
#   Bug 1605641 - Mark 'done' stack value for destructuring assignment as having implicit uses. r=tcampbell
#   
#   We need to make sure this value isn't optimized out because the exception
#   handler may need it.
#   
# ==============================================================================

diff -r 96753184f3c4 -r 6ccaefced3df js/src/vm/Interpreter.h
--- a/js/src/vm/Interpreter.h	Thu Jan 02 12:07:51 2020 +0000
+++ b/js/src/vm/Interpreter.h	Mon Jan 13 09:23:17 2020 +0000
@@ -462,6 +462,18 @@
   const JSTryNote* operator*() const { return tn_; }
 };
 
+class NoOpTryNoteFilter {
+ public:
+  explicit NoOpTryNoteFilter() = default;
+  bool operator()(const JSTryNote*) { return true; }
+};
+
+class TryNoteIterAll : public TryNoteIter<NoOpTryNoteFilter> {
+ public:
+  TryNoteIterAll(JSContext* cx, JSScript* script, jsbytecode* pc)
+      : TryNoteIter(cx, script, pc, NoOpTryNoteFilter()) {}
+};
+
 bool HandleClosingGeneratorReturn(JSContext* cx, AbstractFramePtr frame,
                                   bool ok);
 
